src/crypto/chacha20poly1305/xor.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
```

Comment:
I guess this file is duplicated somewhere, we should considering moving it to an internal package for sharing.